src/main/java/uk/gov/justice/plugin/validator/AnnotationValidator.java

```diff
@@ -8,4 +8,6 @@
 public interface AnnotationValidator {
```

Comment:
Just wondering if we can do this a smarter way and avoid having to implement the method.

Does this look any cleaner?

public interface AnnotationValidator<T> {

   // We can remove this one now...
    String worksOnAnnotatedClass();

    default String getApplicableAnnotationName() {
        return this.getClass().getTypeParameters().getClass().getName();
    }
}